src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -1124,6 +1127,35 @@ void ensureClusterSizeConsistency() {
     }
 
     /**
+     * Verifies that all nodes that have the same version of the cluster state as master have same cluster state
+     */
+    void ensureClusterStateConsistency() throws IOException {
+        if (cluster() != null && isInconsistentClusterStateTest(getTestClass()) == false) {
```

Comment:
instead of having an annotation can we just have a method that we can override? I would like to not make this something people should use.